### DISCUSSION TITLE
Combined diagram wind, windGust, windDir, forecast

### DISCUSSION THREAD

**Discussion by nicholaswaton** [Daveiano/weewx-wdc#236] [Ideas]
Hi there

Thanks again for this wonderful skin! It's been so helpful!

Here's an idea for a combined diagram for your consideration: 
- on one diagram showing the wind, windGusts, wind Directon as an arrow below, and forecasted wind (hour by hour) as a line.
- perhaps showing the current wind as well in a single gauge with wind and wind direction, with gust information.

Attaching an example of what I have in mind.

Obviously I understand that this might be challenging.

<img width="953" alt="Screenshot 2024-01-06 at 10 55 48" src="https://github.com/Daveiano/weewx-wdc/assets/125936903/34e28f1d-60e2-44a1-b6c0-b11586d76ae6">

**Comment by Daveiano** [comment]
Hello there and thank you for your kind words.

Interesting, this one combines current data with forecast data. May I ask where you got this example, any sharable URL? 

Generally, it should be possible to plot all the data provided by the weewx-forecast extension, but I did not get time yet to document this and to create an example, but will do this shortly!

> on one diagram showing the wind, windGusts, wind Directon as an arrow below, and forecasted wind (hour by hour) as a line.

I will think a little bit about this, probably the graph of `windGust`, `windDir` and `windSpeed` is possible with the current skin if you create a custom [`combined_diagram`](https://github.com/Daveiano/weewx-wdc/wiki/Configuration#combined-diagrams). But I will have a look!

> perhaps showing the current wind as well in a single gauge with wind and wind direction, with gust information.

For the other part I could create an optional template in the skin which could be activated via the skin.conf.

Like I said, I will think about this and get back in touch! 
Btw: This is a common use case for sailors, right?

**Comment by nicholaswaton** [comment]
Here's where I got the graph from this site:
 https://www.sailflow.com/spot/153062
or here: https://wx.iwindsurf.com/spot/48645
 They run both premium weather stations where you have to pay, and also some public ones, usually run by sailing clubs.

It's a very common use case for any water sports fan: sailing, kitesurfing, windsurfing, etc.
You basically want to know how the wind has been the last few hours, where it is now, and a projection of how it is going to be, to make decisions about gear, safety, precautions.
